app/frontend/components/common/asset.tsx

```diff
@@ -70,17 +70,25 @@ export const FormattedAssetItem = ({
     }
   }
 
+  const formatAssetName = (metadata: RegisteredTokenMetadata, assetNameHex: string) => {
+    if (metadata?.name) {
```

Comment:
this contradicts the type signature which apparently expects metadata to be defined. If the metadata can actually be null/undefined I'd include it explicitly in the type signature or consider refactoring the upstream code so it doesn't happen. Also, I'd consider passing the assetNameHex first as that value is always defined.

Reply:
Fixed